queries.sql

```diff
@@ -8,3 +8,33 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = TRUE;
 SELECT * from animals WHERE name != 'Gabumon';
 SELECT * from animals WHERE 10.4 <= weight_kg AND weight_kg >= 17.3;
+
+BEGIN;
+ALTER TABLE animals RENAME COLUMN species TO unspecified;
+SELECT * FROM animals;
+ROLLBACK;
+
+-- How many animals are there?
+SELECT COUNT (name) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT (name) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG (weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT MAX (escape_attempts) FROM animals;
+SELECT name FROM animals WHERE escape_attempts >= 7;
+SELECT name FROM animals WHERE neutered = TRUE;
+SELECT name FROM animals WHERE neutered = FALSE;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN (weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT MAX (weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT MIN (weight_kg) FROM animals WHERE species = 'digimon';
+SELECT MAX (weight_kg) FROM animals WHERE species = 'digimon';
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG (escape_attempts) FROM animals WHERE species = 'digimon' AND date_of_birth between '1990-01-01' AND '2000-12-31';
+SELECT AVG (escape_attempts) FROM animals WHERE species = 'pokemon' AND date_of_birth between '1990-01-01' AND '2000-12-31';
```

Comment:
- [ ] Please add a query Inside a transaction that delete all records in the animal's table, then roll back the transaction
```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

Reply:
- [ ] Add a query that
- `DELETE` all animals born after Jan 1st, 2022
- Create a savepoint for the transaction
- `UPDATE` all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction
```sql
BEGIN;

    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```

Reply:
I have forgotten to put those lines in my repos. I was in a rush a little bit...

Reply:
> * [ ]  Please add a query Inside a transaction that delete all records in the animal's table, then roll back the transaction
> 
> ```sql
> BEGIN;
> 
>    DELETE FROM animals WHERE ;
>    SELECT COUNT(*) FROM ANIMALS;
> 
>    ROLLBACK;
>    SELECT COUNT(*) FROM ANIMALS;
> ```
For checking those data we weren't supposed to do so in a transition as long as they don't affect the existing ones. It was only for calculation purposes. so no need to grab the queries within a transition. @DeliceLydia.